File: generated_code/cpp/test_unittest_2532.cpp
#include <gtest/gtest.h>
#include <string>
#include "solution_proxy.h"

// Test Case: Example 1 from the problem description ("abcc" -> true)
TEST(EqualFrequencyTest, Example1_RemoveOneOfDuplicate) {
    Solution solution;
    std::string word = "abcc";
    EXPECT_TRUE(solution.equalFrequency(word));
}

// Test Case: Example 2 from the problem description ("aazz" -> false)
TEST(EqualFrequencyTest, Example2_BalancedFrequencies) {
    Solution solution;
    std::string word = "aazz";
    EXPECT_FALSE(solution.equalFrequency(word));
}

// Edge Case: Minimum constraints (String length 2) - same characters ("aa" -> true)
TEST(EqualFrequencyTest, EdgeCase_MinLength_SameChars) {
    Solution solution;
    std::string word = "aa";
    EXPECT_TRUE(solution.equalFrequency(word));
}

// Edge Case: Minimum constraints (String length 2) - distinct characters ("ab" -> true)
TEST(EqualFrequencyTest, EdgeCase_MinLength_DistinctChars) {
    Solution solution;
    std::string word = "ab";
    EXPECT_TRUE(solution.equalFrequency(word));
}

// Edge Case: All characters have a frequency of 1 ("abcdefg" -> true)
TEST(EqualFrequencyTest, AllUniqueCharacters) {
    Solution solution;
    std::string word = "abcdefg";
    EXPECT_TRUE(solution.equalFrequency(word));
}

// Edge Case: One character appears once, all others appear multiple times equally ("abbcc" -> true)
TEST(EqualFrequencyTest, RemoveSingleOccurrenceCharacter) {
    Solution solution;
    std::string word = "abbcc";
    EXPECT_TRUE(solution.equalFrequency(word));
}

// Edge Case: One character appears N+1 times, all others appear N times ("aabbccc" -> true)
TEST(EqualFrequencyTest, RemoveFromHighestFrequency) {
    Solution solution;
    std::string word = "aabbccc";
    EXPECT_TRUE(solution.equalFrequency(word));
}

// Edge Case: Impossible configuration where removing one doesn't balance the rest ("aaabbbccc" -> false)
TEST(EqualFrequencyTest, BalancedButHigherThanTwo) {
    Solution solution;
    std::string word = "aaabbbccc";
    EXPECT_FALSE(solution.equalFrequency(word));
}

// Edge Case: Single character type repeating multiple times ("aaaaa" -> true)
TEST(EqualFrequencyTest, SingleCharacterRepeating) {
    Solution solution;
    std::string word = "aaaaa";
    EXPECT_TRUE(solution.equalFrequency(word));
}

// Edge Case: Frequencies are close but mathematically impossible to balance ("aabbccddde" -> false)
TEST(EqualFrequencyTest, MultipleFrequencyAnomalies) {
    Solution solution;
    std::string word = "aabbccddde";
    EXPECT_FALSE(solution.equalFrequency(word));
}

File: generated_code/python/test_unittest_2532.py
import os
import importlib.util
import unittest

# Dynamic loading of the Solution class
solution_path = os.environ.get("TEST_SOLUTION_FILE")
spec = importlib.util.spec_from_file_location("Solution", solution_path)
sol_module = importlib.util.module_from_spec(spec)
spec.loader.exec_module(sol_module)
Solution = sol_module.Solution

class TestEqualFrequency(unittest.TestCase):
    def setUp(self):
        self.sol = Solution()

    def test_example_1_abcc(self):
        """Example 1: Removing one 'c' from 'abcc' leaves frequencies {a:1, b:1, c:1}."""
        self.assertTrue(self.sol.equalFrequency("abcc"))

    def test_example_2_aazz(self):
        """Example 2: Removing any char from 'aazz' leaves unequal frequencies (1 and 2)."""
        self.assertFalse(self.sol.equalFrequency("aazz"))

    def test_single_character(self):
        """Constraint: Minimum length string. Removing the only char leaves an empty set (equal)."""
        self.assertTrue(self.sol.equalFrequency("a"))

    def test_all_unique_characters(self):
        """Case: All characters unique. Removing any keeps all frequencies at 1."""
        self.assertTrue(self.sol.equalFrequency("abcdef"))

    def test_all_same_characters(self):
        """Case: All characters the same. Removing one keeps all (remaining) frequencies the same."""
        self.assertTrue(self.sol.equalFrequency("aaaaa"))

    def test_one_extra_frequency_of_one(self):
        """Case: One character has frequency 1, others have frequency N. Remove the unique one."""
        self.assertTrue(self.sol.equalFrequency("aabbccz"))

    def test_one_character_slightly_higher_frequency(self):
        """Case: All but one char have frequency N, one has frequency N+1. Remove from the N+1 one."""
        self.assertTrue(self.sol.equalFrequency("aaabbbcccdddd"))

    def test_two_characters_frequency_two(self):
        """Case: Two characters with frequency 2. Removing one leaves 1 and 2."""
        self.assertFalse(self.sol.equalFrequency("aabb"))

    def test_complex_false_case(self):
        """Case: Multiple differences that cannot be solved with one removal."""
        self.assertFalse(self.sol.equalFrequency("aaabbbccc"))

    def test_max_constraint_valid(self):
        """Case: Large input string where one removal satisfies the condition."""
        word = "a" * 49 + "b" * 50
        self.assertTrue(self.sol.equalFrequency(word))

    def test_single_char_freq_one_remaining_freq_large(self):
        """Case: "zzzzza" -> removing 'a' leaves only 'z's."""
        self.assertTrue(self.sol.equalFrequency("zzzzza"))

    def test_reduction_to_one_not_possible(self):
        """Case: "aabbcd" -> frequencies {2,2,1,1}. Removing one does not equalize others."""
        self.assertFalse(self.sol.equalFrequency("aabbcd"))